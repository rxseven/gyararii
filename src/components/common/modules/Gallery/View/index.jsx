import React, { useState } from 'react';
import { isMobile, isMobileOnly } from 'react-device-detect';
import styled from 'styled-components';

import Button from 'components/common/base/Button';
import Confirm from 'components/common/composite/Confirm';
import Toggle from 'components/common/composite/Toggle';
import Control from 'components/common/sections/Control';
import Error from 'components/common/composite/Error';
import File from 'components/common/composite/File';
import Lightbox from 'components/common/base/Lightbox';
import Jumper from 'components/common/base/Jumper';
import Masonry from 'components/common/sections/Masonry';
import Pagination from 'components/common/composite/Pagination';
import Toolbar from 'components/common/sections/Toolbar';

const Actions = styled.div`
  align-items: center;
  display: ${props => (props['data-visibility'] ? 'flex' : 'none')};
`;

const Autoscroll = styled.div`
  display: ${props => (props['data-visibility'] ? 'block' : 'none')};
`;

const Content = styled.div`
  align-items: center;
  display: flex;
  justify-content: ${({ narrow }) => (narrow ? 'flex-end' : 'space-between')};
`;

const Frame = styled.div`
  margin-top: ${({ isLoading }) => (isLoading ? '-1.5rem' : '36px')};
`;

function View(props) {
  const [isSelecting, setSelecting] = useState(false);
  const {
    action,
    autoscroll,
    currentImage,
    error,
    images,
    isDeleting,
    isFetching,
    isLightbox,
    isModal,
    isPreloading,
    isRestoring,
    isUploading,
    onAutoscroll,
    onDeleteCancel,
    onDeleteConfirm,
    onDeleteRequest,
    onDeselect,
    onDismiss,
    onLightboxClose,
    onLightboxNext,
    onLightboxOpen,
    onLightboxPrev,
    onLoad,
    onLoaded,
    onScrollTop,
    onSelect,
    onSelectAll,
    onUpload,
    pagination,
    selected,
    status
  } = props;
  const errorLength = error.length;
  const imageLength = images.length;
  const selectedLength = selected.length;
  const isLoading = isDeleting || isFetching || isUploading;
  const isMore = !!pagination;
  const isPrefetching =
    (isFetching && isPreloading) || (isRestoring && !imageLength);

  function handleSelecting() {
    setSelecting(!isSelecting);
  }

  return (
    <Frame isLoading={isPrefetching}>
      <Toolbar isLoading={isPrefetching}>
        <Content narrow={!!selectedLength}>
          <Actions data-visibility={!selectedLength}>
            <Button.Set>
              <File
                data-place="right"
                data-tip="Please DO NOT upload inappropriate images."
                data-tip-disable={isMobile}
                icon="cloud-upload-alt"
                id="gallery"
                isLoading={isLoading}
                multiple
                onUpload={onUpload}
                spinner={isUploading}
              >
                Upload
              </File>
              <If condition={isMobile}>
                <Button
                  icon="check"
                  isLoading={isLoading}
                  look={isSelecting ? 'primary' : 'secondary'}
                  onClick={handleSelecting}
                >
                  Tick
                </Button>
              </If>
              <Autoscroll data-visibility={!isMobileOnly}>
                <Toggle
                  checked={autoscroll}
                  isLoading={isLoading}
                  onToggle={onAutoscroll}
                >
                  Auto scroll
                </Toggle>
              </Autoscroll>
            </Button.Set>
          </Actions>
        </Content>
        <Control
          error={error}
          images={images}
          isLoading={isLoading}
          onDeleteRequest={onDeleteRequest}
          onDeselect={onDeselect}
          onError={onScrollTop}
          onSelectAll={onSelectAll}
          selected={selected}
          status={status}
        />
      </Toolbar>
      <Error error={error} onDismiss={onDismiss} />
      <Masonry
        isDeleting={isDeleting}
        isSelecting={isSelecting}
        error={errorLength}
        isFetching={isFetching}
        isMore={isMore}
        images={images}
        onPreview={onLightboxOpen}
        onLoaded={onLoaded}
        onSelect={onSelect}
        selected={selected}
        toDelete={action === 'delete'}
      />
      <Pagination
        isLoading={isFetching}
        isMore={!!imageLength && isMore}
        onLoad={onLoad}
      >
        More images
      </Pagination>
      <Confirm
        buttonConfirm="Delete"
        isOpen={isModal}
        isLoading={isDeleting}
        onClose={onDeleteCancel}
        onConfirm={onDeleteConfirm}
        title="Permanently Delete Images"
        type="danger"
      >
        <p>
          This action is permanent. Are you sure you donâ€™t want to reconsider?
        </p>
      </Confirm>
      <Lightbox
        currentImage={currentImage}
        images={images.map(image => ({ src: image.secure_url }))}
        isOpen={isLightbox}
        onClickNext={onLightboxNext}
        onClickPrev={onLightboxPrev}
        onClose={onLightboxClose}
      />
      <Jumper />
    </Frame>
  );
}

export default View;
