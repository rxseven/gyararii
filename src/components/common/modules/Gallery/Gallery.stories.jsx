import React from 'react';
import LinkTo from '@storybook/addon-links/react';
import { storiesOf } from '@storybook/react';

import notes from 'stories/shared/notes';
import Footnote from 'stories/utilities/Footnote';

import View from './View/View.stories';

const title = 'Components/Common/Modules/Gallery';

storiesOf(title, module)
  .addDecorator(story => (
    <div>
      {story()}
      <Footnote>
        Use it in conjunction with its{' '}
        <LinkTo kind={View.title} story="open" title="Link to View component">
          View
        </LinkTo>{' '}
        component.
      </Footnote>
    </div>
  ))
  .addParameters({
    info: {
      text: `
        **Usage:** use it in conjunction with its <View /> component.
      `
    }
  })
  .add('default', () => <div>A container component, no view later.</div>, {
    notes: notes.minimal
  });

export default { title };
