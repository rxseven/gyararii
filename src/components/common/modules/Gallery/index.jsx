import { has } from 'lodash';
import React from 'react';
import { animateScroll as scroll } from 'react-scroll';
import { isBrowser } from 'react-device-detect';

import withContext from 'HOCs/withContext';

import { isTruthy } from 'helpers/boolean';
import { pluralWord } from 'helpers/string';
import toast from 'helpers/toast';
import * as galelryService from 'services/gallery';

import View from './View';

const MODE = {
  pagination: 'pagination',
  preload: 'preload',
  refill: 'refill'
};

const STATE = {
  ajax: {
    action: null,
    error: [],
    isDeleting: false,
    isFetching: false,
    isPreloading: false,
    isRestoring: false,
    isUploading: false
  },
  gallery: {
    autoscroll: false,
    loaded: false,
    next: 0,
    pagination: null,
    selected: []
  },
  lightbox: {
    currentImage: 0,
    isLightbox: false
  },
  ui: {
    status: null
  }
};

const STATUS_TIMEOUT = 3000;
const UPLOAD_LIMIT_SIZE = 300 * 1000;
const UPLOAD_LIMIT_NUMBER = 10;

class Gallery extends React.Component {
  state = {
    ...STATE.ajax,
    ...STATE.gallery,
    ...STATE.lightbox,
    ...STATE.ui
  };

  isMounting = false;

  loadedImages = 0;

  componentDidMount() {
    const { context } = this.props;
    const { images } = context.state;
    const imagesLength = images.length;

    // Update mounting state
    this.isMounting = true;

    // Fetch images (pleload mode) if the gallery is empty
    if (imagesLength < 1) {
      this.fetchImages({ mode: MODE.preload });
    } else {
      this.setState({ next: imagesLength });
    }
  }

  componentWillUnmount() {
    // Update mounting state
    this.isMounting = false;
  }

  // Update state
  updateState = (updater, callback = null) => {
    if (this.isMounting) {
      this.setState(updater, callback);
    }
  };

  // Delete images
  deleteImages = () => {
    const { context } = this.props;
    const { selected } = this.state;
    const { actions, state } = context;
    const { cursor, images } = state;

    const successHandler = () => {
      // Close a confirmation modal
      actions.modal.onClose();

      // Remove deleted images from the state and reset selected list
      const remaining = images.filter(
        image => !selected.includes(image.public_id)
      );
      const isRefill = cursor && remaining.length < 10;
      const isRestoring = cursor && isTruthy(remaining.length, 0);

      // Update state and status message once the HTTP request finished
      this.handleAjaxSuccess({
        callback: () => {
          // Fetch more images after the previous status message has shown
          if (isRefill) {
            setTimeout(() => {
              this.fetchImages({
                max: selected.length,
                mode: MODE.refill
              });
            }, STATUS_TIMEOUT - (isRestoring ? 0 : 1500));
          }
        },
        gallery: { images: remaining, cursor },
        state: { isRestoring },
        status: `Deleted ${selected.length} ${pluralWord(
          selected.length,
          'image'
        )} successfully`,
        timeout: !isRefill
      });
    };

    const errorHandler = error => {
      // Handle failure HTTP request
      this.handleAjaxError({
        callback: () => {
          // Close modal
          actions.modal.onClose();
        },
        error
      });
    };

    // Update state and status message once the HTTP request starts
    this.handleAjaxStart({ state: { action: 'delete', isDeleting: true } });

    // Remove images on the cloud
    galelryService.deleteImages(selected, successHandler, errorHandler);
  };

  // Fetch images
  fetchImages = (options = { max: null, mode: null }) => {
    const { context } = this.props;
    const { isRestoring } = this.state;
    const { state } = context;
    const { cursor } = state;
    const isMore = isTruthy(options.mode, MODE.pagination);
    const isPreloading = isTruthy(options.mode, MODE.preload);
    const isRefill = isTruthy(options.mode, MODE.refill);
    const request = {
      params: {
        max_results: options.max,
        next_cursor: cursor || null
      }
    };
    let status;

    // Arrange status messages
    switch (true) {
      case isRefill:
        status = 'Updating the gallery...';
        break;
      case isMore:
        status = 'Loading images...';
        break;
      default:
        status = null;
    }

    const successHandler = response => {
      const { images: current } = state;

      // Retrieve data in a response and transform to an appropriate format
      const { data } = response;
      const images = data.resources;

      // Update state and status message once the HTTP request finished
      this.handleAjaxSuccess({
        gallery: {
          cursor: has(data, 'next_cursor') ? data.next_cursor : null,
          images: [...current, ...images]
        },
        state: {
          next: images.length || 0
        },
        status: isMore
          ? `Retrived ${images.length} ${pluralWord(images.length, 'image')}`
          : null
      });
    };

    const errorHandler = error => {
      // Handle failure HTTP request
      this.handleAjaxError({ error, notification: !isPreloading });
    };

    // Clear a selection
    this.handleDeselect();

    // Update state and status message once the HTTP request starts
    this.handleAjaxStart({
      state: {
        isFetching: true,
        isPreloading,
        isRestoring
      },
      status
    });

    // Fetch images from the cloud
    galelryService.fetchImages(request, successHandler, errorHandler);
  };

  // Upload images
  uploadImages = request => {
    const { context } = this.props;
    const { cursor, images: current } = context.state;

    const successHandler = response => {
      // Retrieve data in a response and transform to an appropriate format
      const { data } = response;

      // Insert new uploaded images to the beginning of the existing list
      const images = [...data, ...current];

      // Update state and status message once the HTTP request finished
      this.handleAjaxSuccess({
        gallery: { cursor, images },
        status: `Uploaded ${data.length} ${pluralWord(
          data.length,
          'image'
        )} successfully`
      });
    };

    const errorHandler = error => {
      // Handle failure HTTP request
      this.handleAjaxError({ error });
    };

    // Update state and status message once the HTTP request starts
    this.handleAjaxStart({
      state: { isUploading: true },
      status: 'Uploading images...'
    });

    // Upload images to the cloud
    galelryService.uploadImages(request, successHandler, errorHandler);
  };

  // Handle failure Ajax request
  handleAjaxError = ({ callback = null, error, notification = true }) => {
    // Retrieve error in a response and transform to an appropriate format
    const {
      data: { message }
    } = error.response;

    // Show alert message
    this.updateState(
      { ...STATE.ajax, error: [message], status: null },
      callback
    );

    // Show toast
    if (notification) {
      toast({ message });
    }
  };

  // Handle Ajax start
  handleAjaxStart = ({ state = null, status = null }) => {
    this.updateState({
      ...STATE.ajax,
      ...state,
      status
    });
  };

  // Handle successful Ajax request
  handleAjaxSuccess = ({
    callback = null,
    gallery,
    state = null,
    status = null,
    timeout = true
  }) => {
    const { context } = this.props;

    // Update context state
    if (this.isMounting) {
      context.actions.gallery.onChange(gallery);
    }

    this.updateState(
      {
        ...STATE.ajax,
        ...state,
        pagination: gallery.cursor,
        selected: [],
        status
      },
      callback
    );

    // Automatically remove Ajax status
    if (timeout) {
      setTimeout(() => this.updateState({ status: null }), STATUS_TIMEOUT);
    }
  };

  // Cancel deleting images
  handleDeleteCancel = () => {
    const { context } = this.props;

    // Close a confirmation modal and reset action
    context.actions.modal.onClose();
    this.updateState({ action: null });
  };

  // Confirm deleting images
  handleDeleteConfirm = () => {
    this.deleteImages();
  };

  // Request deleting images
  handleDeleteRequest = () => {
    const { context } = this.props;

    // Open a confirmation modal and set action type
    context.actions.modal.onOpen();
    this.updateState({ action: 'delete' });
  };

  // Fetch more images
  handleLoad = () => {
    this.fetchImages({ mode: MODE.pagination });
  };

  // Check loaded images
  handleLoaded = () => {
    const { context } = this.props;
    const { autoscroll, next } = this.state;

    // Increase a number of loaded images by one
    this.loadedImages = this.loadedImages + 1;

    // When all images loaded successfully
    if (next === this.loadedImages) {
      // Reset counter and tracking state
      this.loadedImages = 0;
      this.updateState({ next: 0, loaded: true });

      // Set pagination
      this.setState({ pagination: context.state.cursor });

      // Scroll to the bottom of the page
      if (autoscroll && isBrowser) {
        scroll.scrollToBottom();
      }
    }
  };

  // Upload images
  handleUpload = event => {
    const errors = [];
    const types = ['image/gif', 'image/jpeg', 'image/png'];

    // Get FileList from a form control
    const fileList = event.target.files;

    // Create a new, shallow-copied Array instance from FileList array-like
    const files = Array.from(fileList);

    // Construct a set of key/value pairs representing form fields and their values
    const formData = new FormData();

    // Check limited number of images
    if (files.length > UPLOAD_LIMIT_NUMBER) {
      // Show an error notification and stop the next execution
      return toast({
        message: `Only ${UPLOAD_LIMIT_NUMBER} images can be uploaded at a time`
      });
    }

    // Check invalid input
    files.forEach((file, index) => {
      // Catching wrong file types
      if (types.every(type => file.type !== type)) {
        errors.push(`${file.type} is not a supported format`);
      }

      // Catching files that are too large
      if (file.size > UPLOAD_LIMIT_SIZE) {
        errors.push(`${file.name} is too large, please pick a smaller file`);
      }

      // Append new key/value to a FormData object
      formData.append(index, file);
    });

    // Show an error notification and stop the next execution
    if (errors.length) {
      return errors.forEach(error => {
        toast({ message: error });
      });
    }

    // Upload images
    return this.uploadImages(formData);
  };

  // Auto-scroll mode
  handleAutoscroll = checked => {
    this.setState(state => ({ autoscroll: !state.autoscroll }));
  };

  // Select image
  handleSelect = id => {
    const { selected } = this.state;
    let selection;

    // Arrange a selection
    if (selected.includes(id)) {
      selection = selected.filter(item => item !== id);
    } else {
      selection = [...selected, id];
    }

    // Update selection list
    this.setState({ selected: selection });
  };

  // Select all images
  handleSelectAll = () => {
    const { context } = this.props;
    const { images } = context.state;

    this.setState({
      selected: images.map(image => image.public_id)
    });
  };

  // Deselect images
  handleDeselect = () => {
    this.setState({ selected: [] });
  };

  // Close lightbox
  handleLightboxClose = () => {
    this.setState({ currentImage: 0, isLightbox: false });
  };

  // Open lightbox
  handleLightboxOpen = index => {
    this.setState({ currentImage: index, isLightbox: true });
  };

  // Show previous image
  handleLightboxPrev = () => {
    const { currentImage } = this.state;

    this.setState({
      currentImage: currentImage - 1
    });
  };

  // Show next image
  handleLightboxNext = () => {
    const { currentImage } = this.state;

    this.setState({
      currentImage: currentImage + 1
    });
  };

  // Dismiss alert
  handleDismiss = alert => {
    const { error } = this.state;

    this.setState({ error: error.filter(item => item !== alert) });
  };

  // Scroll top
  handleScrollTop = () => {
    scroll.scrollToTop();
  };

  render() {
    const { context } = this.props;
    const { state } = this;
    const galleryProps = {
      ...state,
      images: context.state.images,
      isModal: context.state.isModal,
      onAutoscroll: this.handleAutoscroll,
      onDeleteCancel: this.handleDeleteCancel,
      onDeleteConfirm: this.handleDeleteConfirm,
      onDeleteRequest: this.handleDeleteRequest,
      onDeselect: this.handleDeselect,
      onDismiss: this.handleDismiss,
      onLightboxClose: this.handleLightboxClose,
      onLightboxNext: this.handleLightboxNext,
      onLightboxOpen: this.handleLightboxOpen,
      onLightboxPrev: this.handleLightboxPrev,
      onLoad: this.handleLoad,
      onLoaded: this.handleLoaded,
      onScrollTop: this.handleScrollTop,
      onSelect: this.handleSelect,
      onSelectAll: this.handleSelectAll,
      onUpload: this.handleUpload,
      pagination: state.pagination
    };

    return <View {...galleryProps} />;
  }
}

// Export modules
export default withContext(Gallery);
