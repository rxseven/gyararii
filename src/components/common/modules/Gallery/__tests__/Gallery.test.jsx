import { shallow } from 'enzyme';
import React from 'react';
// import renderer from 'react-test-renderer';

import Gallery from '../index';

describe('<Gallery />', () => {
  // Arrange
  const props = {
    // Toolbar
    nodeTop: React.createRef(),

    // File
    id: 'gallery',
    onUpload: jest.fn(),

    // Control
    onDeleteRequest: jest.fn(),
    onDeselect: jest.fn(),
    onScrollTop: jest.fn(),
    onSelectAll: jest.fn(),

    // Error
    error: [],
    onDismiss: jest.fn(),

    // Masonry
    images: [
      {
        public_id: 'image-001',
        secure_url: 'https://via.placeholder.com/480x320?text=Image-001',
        url: 'https://via.placeholder.com/480x320?text=Image-001'
      },
      {
        public_id: 'image-002',
        secure_url: 'https://via.placeholder.com/480x320?text=Image-002',
        url: 'https://via.placeholder.com/480x320?text=Image-002'
      },
      {
        public_id: 'image-003',
        secure_url: 'https://via.placeholder.com/480x320?text=Image-003',
        url: 'https://via.placeholder.com/480x320?text=Image-003'
      }
    ],
    onLightboxOpen: jest.fn(),
    onSelect: jest.fn(),

    // Pagination
    isMore: false,
    onLoad: jest.fn(),

    // Confirm
    buttonConfirm: 'delete',
    isModal: false,
    onDeleteCancel: jest.fn(),
    onDeleteConfirm: jest.fn(),
    title: 'Permanently Delete Images',

    // Lightbox
    currentImage: 1,
    onLightboxNext: jest.fn(),
    onLightboxPrev: jest.fn(),
    onLightboxClose: jest.fn()
  };
  const component = <Gallery {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  // describe('Snapshot tests', () => {
  //   it('should render correctly', () => {
  //     // Act
  //     const tree = renderer.create(component).toJSON();

  //     // Assert
  //     expect(tree).toMatchSnapshot();
  //   });
  // });
});
