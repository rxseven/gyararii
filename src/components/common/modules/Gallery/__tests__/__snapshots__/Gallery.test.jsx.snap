// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> Snapshot tests should render correctly 1`] = `
<div
  className="sc-kEYyzF kOVZJO"
>
  <div
    className="sc-eNQAEJ jEDVOo"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="sc-jKJlTe iXSmQl"
          >
            <label
              className="sc-gqjmRU egfDhJ"
              htmlFor="gallery"
            >
              <span
                className="btn btn-primary btn-sm disabled"
                data-place="right"
                data-tip="Please DO NOT upload inappropriate images."
                disabled={true}
                role="button"
              >
                <span
                  className="sc-htpNat iYntvg"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-cloud-upload-alt fa-w-20 "
                    data-icon="cloud-upload-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 640 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4zM393.4 288H328v112c0 8.8-7.2 16-16 16h-48c-8.8 0-16-7.2-16-16V288h-65.4c-14.3 0-21.4-17.2-11.3-27.3l105.4-105.4c6.2-6.2 16.4-6.2 22.6 0l105.4 105.4c10.1 10.1 2.9 27.3-11.3 27.3z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </span>
                <span
                  className="sc-bxivhb iGoLHq"
                >
                  Upload
                </span>
                <span
                  className="sc-bwzfXH iDiJqN"
                >
                  <span>
                    <style>
                       @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
                    </style>
                    <div
                      style={
                        Object {
                          "WebkitAnimationDelay": "-0.32s",
                          "WebkitAnimationDuration": "1.25s",
                          "WebkitAnimationIterationCount": "infinite",
                          "WebkitAnimationName": "brsk-three-bounce",
                          "WebkitAnimationTimingFunction": "ease-in-out",
                          "animationDelay": "-0.32s",
                          "animationDuration": "1.25s",
                          "animationIterationCount": "infinite",
                          "animationName": "brsk-three-bounce",
                          "animationTimingFunction": "ease-in-out",
                          "backgroundColor": "#fff",
                          "borderRadius": "100%",
                          "display": "inline-block",
                          "height": 7,
                          "marginRight": 5,
                          "width": 7,
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "WebkitAnimationDelay": "-0.16s",
                          "WebkitAnimationDuration": "1.25s",
                          "WebkitAnimationIterationCount": "infinite",
                          "WebkitAnimationName": "brsk-three-bounce",
                          "WebkitAnimationTimingFunction": "ease-in-out",
                          "animationDelay": "-0.16s",
                          "animationDuration": "1.25s",
                          "animationIterationCount": "infinite",
                          "animationName": "brsk-three-bounce",
                          "animationTimingFunction": "ease-in-out",
                          "backgroundColor": "#fff",
                          "borderRadius": "100%",
                          "display": "inline-block",
                          "height": 7,
                          "marginRight": 5,
                          "width": 7,
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "WebkitAnimationDuration": "1.25s",
                          "WebkitAnimationIterationCount": "infinite",
                          "WebkitAnimationName": "brsk-three-bounce",
                          "WebkitAnimationTimingFunction": "ease-in-out",
                          "animationDuration": "1.25s",
                          "animationIterationCount": "infinite",
                          "animationName": "brsk-three-bounce",
                          "animationTimingFunction": "ease-in-out",
                          "backgroundColor": "#fff",
                          "borderRadius": "100%",
                          "display": "inline-block",
                          "height": 7,
                          "width": 7,
                        }
                      }
                    />
                  </span>
                </span>
              </span>
              <input
                className="sc-gZMcBi jOngK"
                disabled={true}
                id="gallery"
                multiple={true}
                onChange={[Function]}
                type="file"
              />
            </label>
            <div
              className="sc-gzVnrw liWgEh"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-hMqMXs dRzBD"
      />
    </div>
  </div>
  <div
    className="sc-kGXeez jhzAZZ"
  >
    <div
      className="card-columns"
    >
      <div />
    </div>
  </div>
  <div
    className="sc-ckVGcZ dRZAJe"
    hidden={false}
  >
    <span>
      <style>
         @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
      </style>
      <div
        style={
          Object {
            "WebkitAnimationDelay": "-0.32s",
            "WebkitAnimationDuration": "1.25s",
            "WebkitAnimationIterationCount": "infinite",
            "WebkitAnimationName": "brsk-three-bounce",
            "WebkitAnimationTimingFunction": "ease-in-out",
            "animationDelay": "-0.32s",
            "animationDuration": "1.25s",
            "animationIterationCount": "infinite",
            "animationName": "brsk-three-bounce",
            "animationTimingFunction": "ease-in-out",
            "backgroundColor": "#999",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": 7,
            "marginRight": 5,
            "width": 7,
          }
        }
      />
      <div
        style={
          Object {
            "WebkitAnimationDelay": "-0.16s",
            "WebkitAnimationDuration": "1.25s",
            "WebkitAnimationIterationCount": "infinite",
            "WebkitAnimationName": "brsk-three-bounce",
            "WebkitAnimationTimingFunction": "ease-in-out",
            "animationDelay": "-0.16s",
            "animationDuration": "1.25s",
            "animationIterationCount": "infinite",
            "animationName": "brsk-three-bounce",
            "animationTimingFunction": "ease-in-out",
            "backgroundColor": "#999",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": 7,
            "marginRight": 5,
            "width": 7,
          }
        }
      />
      <div
        style={
          Object {
            "WebkitAnimationDuration": "1.25s",
            "WebkitAnimationIterationCount": "infinite",
            "WebkitAnimationName": "brsk-three-bounce",
            "WebkitAnimationTimingFunction": "ease-in-out",
            "animationDuration": "1.25s",
            "animationIterationCount": "infinite",
            "animationName": "brsk-three-bounce",
            "animationTimingFunction": "ease-in-out",
            "backgroundColor": "#999",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": 7,
            "width": 7,
          }
        }
      />
    </span>
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "transitionDelay": "0s",
        "transitionDuration": "0.2s",
        "transitionProperty": "opacity, visibility",
        "transitionTimingFunction": "linear",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="sc-jTzLTM iMLVic"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-up fa-w-14 sc-VigVT iSMghy"
        data-icon="arrow-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
